js/lib/jquery.transloadit2.js

```diff
@@ -1,1165 +1,1086 @@
-            canUseCustomBoredLogic = false;
-
-            self._findBoredInstanceUrl(function(err, theUrl) {
-              if (err) {
-                self.ended = true;
-                err = {
-                  error   : 'BORED_INSTANCE_ERROR',
-                  message : self.i18n('errors.BORED_INSTANCE_ERROR') + ' ' + err.message
-                };
-                self.renderError(err);
-                self._options.onError(err);
-                return;
-              }
-
-              url = PROTOCOL + 'api2.' + theUrl + '/instances/bored';
-
-              if (PROTOCOL === 'https://') {
-                url = PROTOCOL + 'api2-' + theUrl + '/instances/bored';
-              }
-
-              proceed();
-            });
-            return;
-          }
+    this._$inputs.on('change', function () {
+      var $input = $(this)
+      self._updateInputFileSelection($input)
+      self._options.onFileSelect($input.val(), $input)
 
-          self.ended = true;
+      if(self._options.triggerUploadOnFileSelection) {
+        self._$form.trigger('submit.transloadit')
+      }
+    })
 
-          var reason = 'JSONP bored instance request status: ' + status;
-          reason += ', err: ' + jsonpErr;
+    this.includeCss()
+  }
 
-          var err = {
-            error   : 'CONNECTION_ERROR',
-            message : self.i18n('errors.CONNECTION_ERROR'),
-            reason  : reason,
-            url     : url
-          };
-          self.renderError(err);
-          self._options.onError(err);
-        }
-      });
+  Uploader.prototype.start = function () {
+    this._xhr = null
+    this._started = false
+    this._ended = false
+    this._pollRetries = 0
+    this._fileCount = 0
+    this._filesSize = 0
+    this._uploadedBytes = 0
+    this._uploads = []
+    this._resumableUploads = []
+    this._uploadFileIds = []
+    this._resultFileIds = []
+    this._results = {}
+
+    this._modal.reset()
+
+    var self = this
+    var cb = function () {
+      setTimeout(function () {
+        self._poll()
+      }, 300)
     }
 
-    proceed();
+    this._countAddedFilesAndSizes()
 
-    if (this._options.modal) {
-      this.showModal();
+    // Simply submit the form if we should not process without files
+    if (this._fileCount === 0 && !this._options.processZeroFiles) {
+      if (this._options.beforeStart()) {
+        this.submitForm()
+      }
+      return
     }
-  };
 
-  Uploader.prototype._findBoredInstanceUrl = function(cb) {
-    var region = this._options.region;
-    var domain = 's3';
-
-    if (region !== 'us-east-1') {
-      domain = 's3-' + region;
+    // Run beforeStart callback before doing any heavy lifting
+    if (!this._options.beforeStart()) {
+      return
     }
 
-    var url = PROTOCOL + domain + '.amazonaws.com/infra-' + region;
-    url    += '.transloadit.com/cached_instances.json';
+    if (!this._checkFileCountExceeded()) {
+      return
+    }
```

Comment:
:+1:

Will keep it like that, though, to make it more apparent that we are actually running the function beforeStart() and it can return false to abort, and that it is not simply part of a or case.